be/src/olap/snapshot_manager.cpp

```diff
@@ -466,44 +467,31 @@ OLAPStatus SnapshotManager::_create_snapshot_files(
             new_tablet_meta->revise_inc_rs_metas(empty_rowsets);
             new_tablet_meta->revise_rs_metas(rs_metas);
         }
-        if (snapshot_version < PREFERRED_SNAPSHOT_VERSION) {
-            set<string> exist_old_files;
-            Status ret = FileUtils::list_dirs_files(schema_full_path, nullptr, &exist_old_files,
-                                       Env::Default());
-            if (!ret.ok()) { 
-                res = OLAP_ERR_DISK_FAILURE;
-                LOG(WARNING) << "failed to dir walk when convert old files. dir=" 
-                             << schema_full_path << ", error:" << ret.to_string();
-                break;
-            }
-            OlapSnapshotConverter converter;
-            TabletMetaPB tablet_meta_pb;
-            OLAPHeaderMessage olap_header_msg;
-            new_tablet_meta->to_meta_pb(&tablet_meta_pb);
-            res = converter.to_old_snapshot(tablet_meta_pb, schema_full_path, schema_full_path, &olap_header_msg);
-            if (res != OLAP_SUCCESS) {
-                break;
-            }
-            // convert new version files to old version files successuflly, then should remove the old files
-            vector<string> files_to_delete;
-            for (auto file_name : exist_old_files) {
-                string full_file_path = schema_full_path + "/" + file_name;
-                files_to_delete.push_back(full_file_path);
-            }
-            // remove all files
-            ret = FileUtils::remove_paths(files_to_delete);
-            if (!ret.ok()) {
-                res = OLAP_ERR_IO_ERROR;
-                LOG(WARNING) << "remove paths failed. error: " << ret.to_string(); 
-                break;
+        if (snapshot_version < BETA_ROWSET_VERSION) {
+            if (snapshot_version == ALPHA_ROWSET_VERSION) {
```

Comment:
this should be following, we must make it fail when meeting unsupported version.
```
if (snapshot_version < ALPHA_ROWSET_VERSION) {
    reurn error;
} else if (snapshot_version == ALPHA_ROWSET_VERSION) {
} else if (snapshot_version < BETA_ROWSET_VERSION) {
} else {
   return error;
}
```